### PR TITLE
VideoPlayer - PVR: fix crash on channel switch for backends not navin…

### PR DESCRIPTION
**Pull request by FernetMenta**
fix trac http://trac.kodi.tv/ticket/16261

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp [hunk 1]
```diff
@@ -383,9 +387,7 @@ bool CDVDInputStreamPVRManager::CloseAndOpen(const char* strFile)
   return false;
 }
 
-bool CDVDInputStreamPVRManager::SupportsChannelSwitch(void)
+bool CDVDInputStreamPVRManager::IsOtherStreamHack(void)
 {
-  PVR_CLIENT client;
-  return g_PVRClients->GetPlayingClient(client) &&
-         client->HandlesInputStream();
+  return m_isOtherStreamHack;
```
Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: just bumped into this PR while checking something else on github ;-) but why don't you just `return m_pOtherStream != NULL;`? looks like you don't need the extra bool for this.

also, this should really be split up into 3 commits (rename method, deleting demux in player, cosmetics) :)

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h [hunk 2]
```diff
@@ -77,6 +77,7 @@ class CDVDInputStream
     virtual bool Record(bool bOnOff) = 0;
     virtual bool CanPause() = 0;
     virtual bool CanSeek() = 0;
+    virtual bool IsOtherStreamHack() = 0;
```
Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: looks like this needs a default `{ return false; }`?

Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: since you've only implemented it for 1 inputstream

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: I intend to remove this generalization. IChannel is only used by PVR

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp (hunk 1 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: > but why don't you just return m_pOtherStream != NULL;? 

not safe, open may fail which can result in this condition too

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: writing this I noticed that I set m_isOtherStreamHack at the wrong place. It needs to be set when trying to open otherStream

Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: which case am I missing?

* no other stream: m_pOtherStream = NULL, m_isOtherStreamHack = false
* input stream failed to create: m_pOtherStream = NULL, m_isOtherStreamHack = false
* input stream failed to open: m_pOtherStream is deleted and NULLed, m_isOtherStreamHack = false
* other stream opened: m_pOtherStream != NULL, m_isOtherStreamHack = true

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h (hunk 2 above)
Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: right yeah, misleading github, didn't see that it was only in IChannel

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp (hunk 1 above)
Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp]: right, crosspost. just noticed that the results were all the same in the current code ;-)

File: xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h (hunk 2 above)
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: this is a method of IChannel :)

```
virtual PVR::CPVRChannelPtr GetSelectedChannel() { return PVR::CPVRChannelPtr(); };
```

Comment (opdenkamp) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: yeah github showed `class CDVDInputStream` above the the change, I'm stupid ;-)

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStream.h]: you are not, we all stumbled into this already :)

**Comment by FernetMenta** [comment]
@opdenkamp I moved assignment of m_isOtherStreamHack up and added a second commit which demotes IChannel interface.
Are you ok with leaving cosmetics in the first commit? Otherwise I would need to write that again :(

**Comment by opdenkamp** [comment]
Sure, you're doing most of the work in /cores/ now anyway, so it's pretty much your choice ;-) I would have cleaned it up before merging though cause this ain't pretty if you need to bisect, revert, etc.

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**